src/modules/household/hooks/useChangeRelationshipToHoh.tsx

```diff
@@ -0,0 +1,107 @@
+import { Stack, Typography } from '@mui/material';
+import { useCallback, useMemo, useState } from 'react';
+
+import ConfirmationDialog from '@/components/elements/ConfirmationDialog';
+import {
+  emptyErrorState,
+  ErrorState,
+  partitionValidations,
+} from '@/modules/errors/util';
+import { getRequiredLabel } from '@/modules/form/components/RequiredLabel';
+import { clientBriefName } from '@/modules/hmis/hmisUtil';
+import RelationshipToHohSelect from '@/modules/household/components/elements/RelationshipToHohSelect';
+import {
+  HouseholdClientFieldsFragment,
+  RelationshipToHoH,
+  useUpdateRelationshipToHoHMutation,
+} from '@/types/gqlTypes';
+
+// hook with logic for changing the HoH in a dialog
+export function useChangeRelationshipToHoh() {
+  // HouseholdClient who's relationship is being changed
+  const [member, setMember] = useState<HouseholdClientFieldsFragment>();
+  // Proposed Relationship to HoH
+  const [relationship, setRelationship] = useState<RelationshipToHoH | null>(
+    null
+  );
+  // This component handles validation errors even though the mutation doesn't currently return any.
+  // This is to prepare for when it does, for example to warn if specifying an adult client as "child".
+  const [errorState, setErrorState] = useState<ErrorState>(emptyErrorState);
+  // Open dialog for specified member
+  const openChangeRelationshipDialog = useCallback(
+    (hc: HouseholdClientFieldsFragment) => {
+      setMember(hc);
+      setRelationship(hc.relationshipToHoH || null);
+    },
+    []
+  );
+
+  const resetState = useCallback(() => {
+    setMember(undefined);
+    setRelationship(null);
+    setErrorState(emptyErrorState);
+  }, []);
+
+  const [updateRelationship, { loading, error }] =
+    useUpdateRelationshipToHoHMutation({
+      onCompleted: (data) => {
+        const errors = data.updateRelationshipToHoH?.errors || [];
+        if (errors.length > 0) {
+          setErrorState(partitionValidations(errors));
+        } else {
+          // on success, reset and close dialog
+          // no need to refetch household because cache is updated
+          resetState();
+        }
+      },
+    });
+
+  const handleConfirm = useCallback(() => {
+    if (!member || !relationship) return;
+    updateRelationship({
+      variables: {
+        input: {
+          enrollmentId: member.enrollment.id,
+          enrollmentLockVersion: member.enrollment.lockVersion,
+          relationshipToHoH: relationship,
+          confirmed: errorState.warnings.length > 0,
```

Comment:
Could you add a comment here explaining why we set confirmed to `errorState.warnings.length > 0`? Is it because this is in the `handleConfirm` callback, so if we had any warnings, the user has now confirmed them; but if there weren't any warnings then we pass `false` since there was nothing to confirm? Is it related to your comment above about handling validations even though the mutation doesn't return any?

Reply:
Good callout. It was something along the lines of "If there are warnings present, that means we already performed the initial request, so pass `confirmed: true` to confirm the warnings and perform the operation." 

But that's not actually a good way to do it, because the user could have changed the relationship on the second request, which might have different warnings associated with it.

I'm just going to remove this. (So, reverting to old behavior where changing rel to hoh doesn't support displaying validations)